### DISCUSSION TITLE
Nested `useFieldArray` validation validates deleted indices.

### DISCUSSION THREAD

**Discussion by DarryQueen** [react-hook-form/react-hook-form#10260] [Q&A]
### Version Number

7.43.9

### Codesandbox/Expo snack

https://codesandbox.io/s/modest-pond-l24ue7

### Steps to reproduce

1. Notice there are two field top-level array fields (`topLevel1` and `topLevel2`).
2. Each top-level array field has a nested array field (`nested1`).
3. Delete one of the top-level fields.
4. Click submit.
5. The validation error at the nested array level (`` `array.${number}.nestedArray` ``) throws an error, saying the array value is undefined.

### Expected behaviour

The validation should succeed (or at least not error).

I believe this is because additional indices are not being accounted for. There are originally 2 array values; I delete one so there should only be 1. The nested validation (`` `array.${index}.nestedArray` ``) is validating with field names `` `array.${0}.nestedArray` `` *and* `` `array.${1}.nestedArray` ``, and the latter cannot find a value (because it was deleted).

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by bluebill1049** [comment]
```
resolver: async (data, context, options) => {
  // you can debug your validation schema here
  console.log('formData', data)
  console.log('validation result', await anyResolver(schema)(data, context, options))
  return anyResolver(schema)(data, context, options)
},
```

please use the above code to bug your issue.

**Comment by leapful** [comment]
@DarryQueen 

Please read carefully the documentation https://react-hook-form.com/api/usefieldarray/

<img width="955" alt="Screenshot 2023-04-12 at 18 40 51" src="https://user-images.githubusercontent.com/18494222/231446351-962706ab-c874-49d9-9b7a-330d56ab266e.png">

`useFieldArray` hook does not support dynamic field array name therefore your nested field `array.0.nestedArray` cannot be found on form field reference and causes undefined data.

**Reply by DarryQueen** [reply]
What does dynamic name mean? Is this something that can be supported soon? `useFieldArray` is a really nice and convenient helper, and nested arrays seem like a fairly common use case.

**Reply by leapful** [reply]
@DarryQueen 

Dynamic name is the field array name you used in this file https://codesandbox.io/s/modest-pond-l24ue7?file=/src/NestedFields.tsx . It's currently not supported at this time.
